suse_migration_services/units/product_setup.py

```diff
@@ -56,73 +49,22 @@ def main():
         # including distro_target. The current workaround solution is
         # to delete the target specification in the baseproduct
         # registration if present.
-        products_metadata = os.sep.join(
-            [root_path, 'etc', 'products.d']
-        )
-        if os.path.exists(products_metadata):
-            log.info('Creating backup of Product data')
-            Command.run(
-                [
-                    'rsync', '-zav', '--delete', products_metadata + os.sep,
-                    '/tmp/products.d.backup/'
-                ]
-            )
-        baseproducts = get_baseproduct(products_metadata)
-        if len(baseproducts) != 1:
-            if not baseproducts:
+        base_products = SUSEProduct().get_baseproduct()
+        if len(base_products) != 1:
+            if not base_products:
                 message = 'There is no baseproduct'
             else:
-                message = ('Found multiple product definitions '
-                           'without element <flavor>: \n{0}'
-                           .format('\n'.join(baseproducts)))
+                message = (
+                    'Found multiple product definitions '
+                    'without element <flavor>: \n{0}'
+                    .format('\n'.join(base_products))
+                )
             log.error(message)
             raise DistMigrationProductSetupException(message)
 
         log.info('Updating Base Product to be suitable for migration')
-        delete_target_registration(baseproducts[0])
+        SUSEProduct().delete_target_registration(base_products[0])
```

Comment:
Do you have any reason why you don't create an instance of SUSEProduct() ?

```python
suse_product = SUSEProduct()
suse_product.get_baseproduct()
suse_product.delete_target_registration(...)
```

Reply:
No reason. I will update it with an instance.